src/main/java/org/cactoos/text/FormattedText.java

```diff
@@ -148,27 +129,22 @@ public FormattedText(
      *
      * @param ptn Pattern
      * @param locale Format locale
-     * @param arguments Arguments
+     * @param args Arguments
      */
     public FormattedText(
         final Text ptn,
         final Locale locale,
-        final Collection<Object> arguments
+        final Collection<Object> args
     ) {
-        this.pattern = ptn;
-        this.locale = locale;
-        this.args = Collections.unmodifiableCollection(arguments);
-    }
-
-    @Override
-    public String asString() throws Exception {
-        final StringBuilder out = new StringBuilder(0);
-        try (final Formatter fmt = new Formatter(out, this.locale)) {
-            fmt.format(
-                this.pattern.asString(),
-                this.args.toArray(new Object[0])
-            );
-        }
-        return out.toString();
+        super((Scalar<String>) () -> {
```

Comment:
@victornoel Since type casting is [discouraged in EO](https://www.elegantobjects.org/), consider adding a private constructor which would take `Scalar<String>` (or just `Text`) and call `super` from there.